R/utils.R

```diff
@@ -6,7 +6,14 @@
 #'
 #' @examples
 get_margins <- function (chars) {
-  return (trimws(strsplit(chars, ":")[[1]]))
+  chars <- trimws(chars)
+  left_is_exclusive <- substr(chars, 1, 1) == "("
```

Comment:
We should use interval terminology which can be reference here https://en.wikipedia.org/wiki/Interval_(mathematics).
For e.g., this variables would be called `is_left_open`.